pandas/io/sql.py

```diff
@@ -728,42 +724,67 @@ def has_table(table_name: str, con, schema: str | None = None) -> bool:
     -------
     boolean
     """
-    pandas_sql = pandasSQL_builder(con, schema=schema)
-    return pandas_sql.has_table(table_name)
+    with pandasSQL_builder(con, schema=schema) as pandas_sql:
+        return pandas_sql.has_table(table_name)
 
 
 table_exists = has_table
 
 
-def pandasSQL_builder(con, schema: str | None = None) -> SQLDatabase | SQLiteDatabase:
+@contextmanager
+def pandasSQL_builder(
+    con,
+    schema: str | None = None,
+    need_txn: bool = False,
```

Comment:
```suggestion
    need_transaction: bool = False,
```

IMO it's a bit clearer.

Reply:
I changed this to `need_transaction`.